providers/common/digest_to_nid.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/objects.h>
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+#include <openssl/core.h>
+#include "prov/check.h"
+#include "internal/nelem.h"
+
+/*
+ * Internal library code deals with NIDs, so we need to translate from a name.
+ * We do so using EVP_MD_is_a(), and therefore need a name to NID map.
+ */
+int digest_md_to_nid(const EVP_MD *md, const OSSL_ITEM *it, size_t it_len)
+{
+    size_t i;
+    int mdnid = NID_undef;
+
+    if (md == NULL)
+        goto end;
+
+    for (i = 0; i < it_len; i++) {
+        if (EVP_MD_is_a(md, it[i].ptr)) {
+            mdnid = (int)it[i].id;
```

Comment:
return last match found is deliberate?  worth a comment before the for.  Also in that comment perhaps explain why linear search is okay.

Reply:
It breaks when it finds one..
I don't think a comment is necessary.

Reply:
oops, sorry, I missed the "break" and you're right a comment isn't necessary.
I would rewrite the loop like this:
```
for (i = 0; i < it_len; i++, it++)
   if (EVP_MD_is_ad(md, it->ptr))
        return (int)it->id;
return NID_undef;
```
and get rid of the mdnid variable.

Reply:
done